src/ert/run_models/everest_run_model.py

```diff
@@ -355,103 +355,181 @@ def _run_forward_model(
         self,
         control_values: NDArray[np.float64],
         model_realizations: list[int],
-        active_control_vectors: list[bool] | None = None,
+        active_sim_ids: list[bool] | None = None,
     ) -> tuple[NDArray[np.float64], NDArray[np.float64] | None, list[int]]:
         # Reset the current run status:
         self._status = None
 
         # Get cached_results:
-        cached_results = self._get_cached_results(control_values, model_realizations)
+        cached_results_by_simid = self._get_cached_results_by_sim_id(
+            control_values, model_realizations
+        )
 
         # Collect the indices of the controls that must be evaluated in the batch:
-        evaluated_control_indices = [
-            idx
-            for idx in range(control_values.shape[0])
-            if idx not in cached_results
-            and (active_control_vectors is None or active_control_vectors[idx])
+        sim_ids_to_evaluate = [
```

Comment:
I don't like this indentation tbh. Could you instead just do:
```ptyhon
if not sim_ids_to_evaluate:
self._batch_id += 1

        # Return the results, together with the indices of the evaluated controls:
        return objectives, constraints, sim_ids_to_evaluate
```
and then just continue?